### DISCUSSION TITLE
SQL size limit for Dataset

### DISCUSSION THREAD

**Discussion by spartyann** [apache/superset#18844] [Ideas]
**Is your feature request related to a problem? Please describe.**
Hello,
Sorry for my bad English I am French :)
We have encountered a problem when using your API. We wanted to create datasets but one of our datasets brings a SQL query with a size of 82.000. Yes I know it's big but we absolutely need this size because the SQL is generated dynamically from the configuration of our customers.

**Describe the solution you'd like**
In the Superset database the "tables.sql" column is of type "text" which limits the size to 65,535 in Mariadb. So I switched it to "mediumtext" and it works.

But it's not a good thing to modify Superset tables.
If you could add this migration to Superset I would be most happy!

I did this:
`ALTER TABLE tables CHANGE sql sql mediumtext NULL;`

Thanks very much !!
Yann

**Comment by villebro** [comment]
Converting to discussion, as this is an edge case.
